Facter 3.9 documentation (HOLD FOR RELEASE)

gguillotte opened the pull request:
-   Add Facter 3.9 release notes.
-   Update core facts list.

Relevant files to review:

See commit 379735845c33549e1e2adefcd07e75fe0968d8b7.

-   core_facts.md
-   release_notes.md

Changes still pending for this PR:

None. References have already been generated from facter commit [`1650f44`](https://github.com/puppetlabs/facter/commit/1650f446b4b6076ab0e5937ef9e47a4523fc385b), which includes the new `hypervisor` fact.

Changes to be made in other PRs:

-   Puppet 5.2 release notes and docs (PR #785)
-   Puppet Server 5.1 release notes and docs (puppet-server [PR #1506](https://github.com/puppetlabs/puppetserver/pull/1506))
-   Hiera release notes (if any)
-   `_config.yml` updates to publish the docs (PR #787)

puppetcla commented:
CLA signed by all contributors.

gguillotte commented:
FYI @caseywilliams @branan, if you want to take a look.

caseywilliams commented on source/facter/3.9/release_notes.md:
```diff
@@ -1,44 +1,27 @@
-* Virtualization detection for FreeBSD Jails has been added.
-
-* Backward support added for zpool feature flags prior to verion 28.
+    The new fact recognizes multiple hypervisors in nested virtualization environments, and includes metadata about each hypervisor where available. The `hypervisor` fact and libwhereami are the first steps toward improved virtualzation support in future releases of Facter. They also reduce its dependence on the external tool `virt-what`, which has a few detection bugs and requires root to run.
```
Small change: this should be `hypervisors`

gguillotte commented:
Addressed feedback.

FYI, I am on PTO and without internet access starting now and going through tomorrow. For further changes to these docs, please merge this PR and request a follow-up docs PR. @MFredette is the docs contact for this release.
